tests/test_fetch.py

```diff
@@ -21,39 +20,53 @@
 
 @pytest.mark.integration
 class TestConnectionWithShortTimeouts(TestCase):
-
     def test_download_connectionerror(self):
-        with env_var('CONDA_REMOTE_CONNECT_TIMEOUT_SECS', 1, stack_callback=conda_tests_ctxt_mgmt_def_pol):
-            with env_var('CONDA_REMOTE_READ_TIMEOUT_SECS', 1, stack_callback=conda_tests_ctxt_mgmt_def_pol):
-                with env_var('CONDA_REMOTE_MAX_RETRIES', 1, stack_callback=conda_tests_ctxt_mgmt_def_pol):
-                    with pytest.raises(CondaHTTPError) as execinfo:
-                        url = "http://240.0.0.0/"
-                        msg = "Connection error:"
-                        download(url, mktemp())
-                        assert msg in str(execinfo)
+        with env_vars(
+            {
+                "CONDA_REMOTE_CONNECT_TIMEOUT_SECS": 1,
+                "CONDA_REMOTE_READ_TIMEOUT_SECS": 1,
+                "CONDA_REMOTE_MAX_RETRIES": 1,
+            },
+            stack_callback=conda_tests_ctxt_mgmt_def_pol,
+        ):
+            with pytest.raises(CondaHTTPError) as execinfo:
```

Comment:
Might as well chain the contexts too:

```suggestion
        with env_vars(
            {
                "CONDA_REMOTE_CONNECT_TIMEOUT_SECS": 1,
                "CONDA_REMOTE_READ_TIMEOUT_SECS": 1,
                "CONDA_REMOTE_MAX_RETRIES": 1,
            },
            stack_callback=conda_tests_ctxt_mgmt_def_pol,
        ), pytest.raises(CondaHTTPError) as execinfo:
```